src/test/java/com/redhat/cloud/notifications/routers/NotificationServiceTest.java

```diff
@@ -145,4 +146,133 @@ void testNonExistantDefaults() {
                 .then()
                 .statusCode(400);
     }
+
+    @Test
+    void testGetEventTypesAffectedByEndpoint() {
+        String tenant = "testGetEventTypesAffectedByEndpoint";
+        String userName = "user";
+        String localIdentityHeaderValue = TestHelpers.encodeIdentityInfo(tenant, userName);
+        Header localIdentityHeader = TestHelpers.createIdentityHeader(localIdentityHeaderValue);
+        mockServerConfig.addMockRbacAccess(localIdentityHeaderValue, MockServerClientConfig.RbacAccess.FULL_ACCESS);
+
+        List<Application> applications = this.helpers.getApplications();
+        UUID ep1 = this.helpers.createWebhookEndpoint(tenant);
+        UUID ep2 = this.helpers.createWebhookEndpoint(tenant);
+        UUID defaultEp = this.helpers.getDefaultEndpointId(tenant);
+        List<EventType> eventTypesFromApp1 = this.helpers.getEventTypesForApplication(applications.get(1).getId());
+        EventType ev0 = eventTypesFromApp1.get(0);
+        EventType ev1 = eventTypesFromApp1.get(1);
+
+        // ep1 assigned to ev0; ep2 not assigned; default not assigned.
+        this.helpers.assignEndpointToEventType(tenant, ep1, ev0.getId());
+        Response response = given()
+                .header(localIdentityHeader)
+                .when()
+                .contentType(ContentType.JSON)
+                .get("/notifications/eventTypes/affectedByRemovalOfEndpoint/" + ep1.toString())
+                .then()
+                .statusCode(200)
+                .extract().response();
+
+        EventType[] eventTypes = Json.decodeValue(response.getBody().asString(), EventType[].class);
```

Comment:
Just cosmetic and I also need to look that up every time:

you could do something like `String jsonBody = ... .extract().response().bodyAsString(); above and then directly decode here.